### PR TITLE
Change CI settings for support Ruby3.0+ Rails6.1+ 

### LINKED ISSUE #340: Deprecate Ruby 2.7
Per https://github.com/ossf/osv-schema/issues/123#issuecomment-1490334983, Ruby 2.7 has reached end-of-life, and the new minimum version should be at least 3.0 which is in security-only maintenance.

Need to update the CI and gemspec accordingly.

### PR DESCRIPTION
**Pull request by willnet**
Now, Ruby 2.7 is EOL, and Rails 6.0 is also EOL.

So, I changed the CI settings to support only those higher versions and made the changes necessary to pass the tests.

CI result is [here](https://github.com/willnet/sorcery/actions/runs/7724024577).

Fix #340

### CODE REVIEW

File: gemfiles/rails_61.gemfile
Comment (joshbuker) [inline gemfiles/rails_61.gemfile:6]: >= version locks can cause issues, but I'm not concerned enough about it to block the PR.

File: sorcery.gemspec
Comment (joshbuker) [inline sorcery.gemspec:41]: Might be better to just remove the version lock instead, but I vaguely recall dev dependencies in the gemspec being deprecated anyway.

File: spec/controllers/controller_session_timeout_spec.rb
Comment (joshbuker) [inline spec/controllers/controller_session_timeout_spec.rb:134]: Because this touches authentication, I'll need to verify before merging this.

File: spec/controllers/controller_spec.rb
Comment (joshbuker) [inline spec/controllers/controller_spec.rb:69]: Because this touches csrf, I'll need to verify before merging this.

**Review by joshbuker — COMMENTED** [review]
Need to verify before merging.

**Review by joshbuker — APPROVED** [review]
NGL, I can't find any mentions of changes to the csrf functionality that would cause the test to fail in Rails 7.1. Has me pretty spooked, but I don't have the time to investigate further and Rails 7.1 is largely broken in its current state for Sorcery v0 anyway.

Really need to finish that v1 rework, but it's been long enough that getting back into it is challenging.

File: spec/controllers/controller_spec.rb
Comment (willnet) [inline spec/controllers/controller_spec.rb:72]: @joshbuker Sorcery uses the [form_authenticity_token method](https://github.com/Sorcery/sorcery/blob/856eacd9912ecbb436a7259ba927f171d77b0ba1/lib/sorcery/controller.rb#L57) to assign values to `session[:_csrf_token]`. However, it seems that due to [this commit](https://github.com/rails/rails/commit/f2c66ce392016efd49816cd6fdc9f174271a0321) in Rails 7.1, the behavior of the form_authenticity_token method has changed and it no longer assigns values to `session[:_csrf_token]`.

If it's necessary to assign a value to `session[:_csrf_token]` during the execution of the login method, then we may need to revise our implementation. The question arises whether this is actually necessary. Rails seems to assign values to `session[:_csrf_token]` as needed, so there might not be a clear need for Sorcery to explicitly do this. There doesn't seem to be a specific reason mentioned in the commit where [form_authenticity_token was introduced](https://github.com/Sorcery/sorcery/commit/8a7490c3b6b8e64c9b6d092103a74be77e4093e6), making it hard to infer the rationale behind this implementation.

Comment (willnet) [reply spec/controllers/controller_spec.rb:72]: @joshbuker I did some additional research on this topic. The PR that introduced the form_authenticity_token method is [here](https://github.com/NoamB/sorcery/pull/409), and it seems it was merged to fix [this issue](https://github.com/NoamB/sorcery/issues/250) (though it's mentioned as `Fix for #150` in the PR, it seems like a typo and should probably be `Fix for #250` based on the content).

However, calling form_authenticity_token during login doesn't resolve the issue mentioned in https://github.com/NoamB/sorcery/issues/250, making [this commit](https://github.com/Sorcery/sorcery/commit/8a7490c3b6b8e64c9b6d092103a74be77e4093e6) seem rather pointless. Moreover, invoking form_authenticity_token has led to a new issue being created, as seen in [Using sorcery in an API-only rails app does not work · Issue #330 · Sorcery/sorcery](https://github.com/Sorcery/sorcery/issues/330), suggesting that it might be better to remove this implementation and related tests.

**Pull request merged**
